### PR TITLE
Staking proxy list

### PR DESCRIPTION
**Pull request by antonijzelinskij**
#869379hxa

### CODE REVIEW

**Comment by stepanLav** [comment]
Task linked: [CU-869379hxa Android. Your staking proxies screen](https://app.clickup.com/t/869379hxa)

File: feature-proxy-impl/src/main/java/io/novafoundation/nova/feature_proxy_impl/data/repository/RealGetProxyRepository.kt [hunk 1]
```diff
@@ -81,6 +84,25 @@ class RealGetProxyRepository(
...
+            runtime.metadata.module(Modules.PROXY)
+                .storage("Proxies")
+                .observe(
+                    accountId,
+                    binding = { bindProxyAccounts(it) }
+                )
+        }.map {
+            it.proxies.filter { it.proxyType == proxyType.name }
+                .map { ProxiedWithProxy.Proxy(it.accountId.value, it.proxyType) }
+        }
```
Comment (valentunn) [inline feature-proxy-impl/src/main/java/io/novafoundation/nova/feature_proxy_impl/data/repository/RealGetProxyRepository.kt:98]: Haven't we decided to use updater and subscribe to local storage here?

File: feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/staking/delegation/proxy/list/StakingProxyListInteractor.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import io.novafoundation.nova.feature_account_api.domain.model.requireAccountIdIn
+import io.novafoundation.nova.feature_proxy_api.data.repository.GetProxyRepository
+import io.novafoundation.nova.feature_proxy_api.domain.model.ProxyType
+import io.novafoundation.nova.feature_staking_impl.domain.staking.delegation.proxy.list.model.StakingProxyAccount
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import jp.co.soramitsu.fearless_utils.runtime.AccountId
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+
+class StakingProxyListInteractor(
```
Comment (valentunn) [inline feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/staking/delegation/proxy/list/StakingProxyListInteractor.kt:15]: Lets try to always have interface for interactors

File: feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/staking/delegation/proxy/list/StakingProxyListInteractor.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    fun stakingProxyListFlow(chain: Chain, accountId: AccountId): Flow<List<StakingProxyAccount>> {
+        return combine(
+            accountRepository.activeMetaAccountsFlow(),
+            getProxyRepository.proxiesByTypeFlow(chain, accountId, ProxyType.Staking)
+        ) { metaAccounts, proxies ->
+            val accountIdToMetaAccount = metaAccounts
+                .filter { it.hasAccountIn(chain) }
+                .associateBy {
+                    it.requireAccountIdIn(chain).intoKey()
+                }
```
Comment (valentunn) [inline feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/staking/delegation/proxy/list/StakingProxyListInteractor.kt:29]: Maybe we can use `LocalIdentityProvider` here? We just need to implement `identitiesFor` the same way you did here

**Review by valentunn — APPROVED** [review]

**Pull request merged**